src/main.c

```diff
@@ -154,14 +153,15 @@ main (int    argc,
 
   loop = g_main_loop_new (NULL, FALSE);
 
-  sigint_id = 0;
   if (!opt_no_sigint)
     {
-      sigint_id = g_unix_signal_add_full (G_PRIORITY_DEFAULT,
+      guint sigint_id = g_unix_signal_add_full (G_PRIORITY_DEFAULT,
```

Comment:
Please declare this variable at the beginning of the function (i.e. leave the old declaration in place).